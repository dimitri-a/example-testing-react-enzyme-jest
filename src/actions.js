export const handleStartCar = () => {
  console.log('car has started');
};
