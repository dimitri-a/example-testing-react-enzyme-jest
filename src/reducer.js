export default function(state, action) {
  if (action.type === 'HANDLE_START_CAR') {
    return {
      ...state,
      isRunning: true
    };
  }
  return state;
};
