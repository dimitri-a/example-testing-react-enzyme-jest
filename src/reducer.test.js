import reducer from './reducer';
import * as actions from './actions';

describe('reducer', () => {
  describe('HANDLE_START_CAR', () => {
    it('sets isRunning to true', function () {
      const previousState = {
        otherProp: 'test-123'
      };
      const action = actions.handleStartCar();
      const nextState = reducer(previousState, action);
      /* expect(nextState).toEqual({
        isRunning: true,
        otherProp: 'test-123'
      }); */
      // use snapshot instead of boilerplate to check all the properties
      expect(nextState).toMatchSnapshot();
    });
  });
});
