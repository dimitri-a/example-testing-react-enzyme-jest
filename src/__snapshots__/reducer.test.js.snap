exports[`reducer HANDLE_START_CAR sets isRunning to true 1`] = `
Object {
  "isRunning": true,
  "otherProp": "test-123",
}
`;
